drivers/i2c/i2c_sam0.c

```diff
@@ -137,6 +139,17 @@ static void i2c_sam0_isr(const struct device *dev)
 		return;
 	}
 
+	/*
+	 * Directly send/receive next message if it is in the same direction and
+	 * the current message has no stop or restart flags.
```

Comment:
This comment became inaccurate after the recent change regarding the restart flag. It would be nice to update it.